From a134a47c26b95fd6f703629186202f08ef749e77 Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Fri, 4 Dec 2020 09:22:49 -0600
Subject: [PATCH] msvc: fix for linking rtl_biast with libopt

Added biast app to the list of apps that link with libgetopt_static

Also only linked libm with specified libraries when libm is found.
The libm may not be present in msvc and gives a linker error,
although other windows build environment may have it.
---
 src/CMakeLists.txt | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f1907e3..44221d9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -211,9 +211,16 @@ if(WIN32)
     target_link_libraries(rtl_eeprom libgetopt_static)
     target_link_libraries(rtl_adsb libgetopt_static)
     target_link_libraries(rtl_power libgetopt_static)
-    target_link_libraries(rtl_raw2wav libgetopt_static m)
-    target_link_libraries(rtl_wavestat libgetopt_static m)
-    target_link_libraries(rtl_wavestream libgetopt_static m)
+    target_link_libraries(rtl_biast libgetopt_static)
+    target_link_libraries(rtl_raw2wav libgetopt_static)
+    target_link_libraries(rtl_wavestat libgetopt_static)
+    target_link_libraries(rtl_wavestream libgetopt_static)
+    find_library(LIBM NAMES m)
+    if (LIBM)
+    target_link_libraries(rtl_raw2wav m)
+    target_link_libraries(rtl_wavestat m)
+    target_link_libraries(rtl_wavestream m)
+    endif (LIBM)
     set_property(TARGET rtl_sdr APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_tcp APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_udp APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
@@ -223,6 +230,7 @@ if(WIN32)
     set_property(TARGET rtl_eeprom APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_adsb APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_power APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
+    set_property(TARGET rtl_biast APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_raw2wav APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_wavestat APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
     set_property(TARGET rtl_wavestream APPEND PROPERTY COMPILE_DEFINITIONS "rtlsdr_STATIC" )
